packages/lexical-markdown/src/utils.ts

```diff
@@ -460,3 +460,43 @@ export function isEmptyParagraph(node: LexicalNode): boolean {
       MARKDOWN_EMPTY_LINE_REG_EXP.test(firstChild.getTextContent()))
   );
 }
+
+// URL validation patterns
+export const EMAIL_REGEX =
+  /(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))/;
+
+export const PHONE_NUMBER_REGEX = /^\+?[0-9\s()-]{5,}$/;
+
+/**
+ * Formats a URL string by adding appropriate protocol if missing
+ *
+ * @param url - URL to format
+ * @returns Formatted URL with appropriate protocol
+ */
+export function formatUrl(url: string): string {
+  // Check if URL already has a protocol
+  if (url.match(/^[a-z]+:/i)) {
```

Comment:
This doesn't quite match the URI spec https://datatracker.ietf.org/doc/html/rfc3986#section-3.1 (although a browser is not often going to do something useful with something like `s3://` or `ms-excel:` they are still valid)

```suggestion
  if (url.match(/^[a-z][a-z0-9+.-]*:/i)) {
```